packages/execution-environments/src/common/endowments/network.ts

```diff
@@ -0,0 +1,343 @@
+        () => openConnections.delete(openConnection),
+      );
+    }
+
+    /* istanbul ignore next: mock implementation has a bug https://github.com/thoov/mock-socket/issues/242 */
+    get onclose(): WebSocketCallback | null {
+      return this.#oncloseOriginal;
+    }
+
+    set onclose(callback: WebSocketCallback | null) {
+      if (this.#isTornDown !== true) {
+        this.#oncloseOriginal = callback;
+        this.#socket.onclose = this.#createWrapped(callback);
+      }
+    }
+
+    /* istanbul ignore next: mock implementation has a bug https://github.com/thoov/mock-socket/issues/242 */
+    get onerror(): ((this: WebSocket, ev: Event) => any) | null {
+      return this.#onerrorOriginal;
+    }
+
+    set onerror(callback: ((this: WebSocket, ev: Event) => any) | null) {
+      this.#onerrorOriginal = callback;
+      this.#socket.onerror = this.#createWrapped(callback);
+    }
+
+    /* istanbul ignore next: mock implementation has a bug https://github.com/thoov/mock-socket/issues/242 */
+    get onmessage(): ((this: WebSocket, ev: MessageEvent<any>) => any) | null {
+      return this.#onmessageOriginal;
+    }
+
+    set onmessage(
+      callback: ((this: WebSocket, ev: MessageEvent<any>) => any) | null,
+    ) {
+      this.#onmessageOriginal = callback;
+      this.#socket.onmessage = this.#createWrapped(callback);
+    }
+
+    /* istanbul ignore next: mock implementation has a bug https://github.com/thoov/mock-socket/issues/242 */
+    get onopen(): ((this: WebSocket, ev: Event) => any) | null {
+      return this.#onopenOriginal;
+    }
+
+    set onopen(callback: ((this: WebSocket, ev: Event) => any) | null) {
+      this.#onopenOriginal = callback;
+      this.#socket.onopen = this.#createWrapped(callback);
+    }
+
+    close(code?: number, reason?: string): void {
+      this.#socket.close(code, reason);
+    }
+
+    send(data: string | Blob | ArrayBufferView | ArrayBufferLike): void {
+      this.#socket.send(data);
+    }
+
+    get CLOSED(): number {
+      return this.#socket.CLOSED;
+    }
+
+    get CLOSING(): number {
+      return this.#socket.CLOSING;
+    }
+
+    get CONNECTING(): number {
+      return this.#socket.CONNECTING;
+    }
+
+    get OPEN(): number {
+      return this.#socket.OPEN;
+    }
+
+    get binaryType(): BinaryType {
+      return this.#socket.binaryType;
+    }
+
+    set binaryType(value: BinaryType) {
+      this.#socket.binaryType = value;
+    }
+
+    get bufferedAmount(): number {
+      return this.#socket.bufferedAmount ?? 0;
+    }
+
+    get extensions(): string {
+      return this.#socket.extensions ?? '';
+    }
+
+    get protocol(): string {
+      return this.#socket.protocol;
+    }
+
+    get readyState(): number {
+      return this.#socket.readyState;
+    }
+
+    get url(): string {
+      return this.#socket.url;
+    }
+
+    addEventListener<K extends keyof WebSocketEventMap>(
```

Comment:
So that it works correctly when used in Typescript, in example our tests. Otherwise the expected return values would differ from actual WebSocket implementation and we would have to handle operations differently.

Reply:
So that it works correctly when used in Typescript, in example our tests. Otherwise the expected return values would differ from actual WebSocket implementation and we would have to handle operations differently.

Reply:
So that it works correctly when used in Typescript, in example our tests. Otherwise the expected return values would differ from actual WebSocket implementation and we would have to handle operations differently.